### PR TITLE
chore(website): [playground] improve mobile design

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Improve playground design on mobile devices

this change has been extracted from #6656

### CODE REVIEW

**Review by bradzacher — CHANGES_REQUESTED** [review]
The layout seems a bit wrong on mobile - the bottom panel is real small and I can't scroll to see the content.

**Comment by armano2** [comment]
can you tell me what browser / os is that, i will check it out

File: packages/website/src/components/Playground.tsx [hunk 1]
```diff
@@ -60,133 +74,151 @@ function Playground(): React.JSX.Element {
     }
   }, []);
 
+  useEffect(() => {
+    if (windowSize === 'mobile') {
+      playgroundMenuRef.current?.collapse();
+    } else if (windowSize === 'desktop') {
+      playgroundMenuRef.current?.expand();
+    }
+  }, [windowSize, playgroundMenuRef]);
```
Comment (StyleShit) [inline packages/website/src/components/Playground.tsx:83]: why do you need the ref as a dependency here? doesn't it keep the same reference on each re-render anyway?

```suggestion
  }, [windowSize]);
```

File: packages/website/src/components/Playground.tsx [hunk 2]
```diff
@@ -60,133 +74,151 @@ function Playground(): React.JSX.Element {
...
+                  filter={state.showAST === 'es' ? esQueryFilter : undefined}
+                  value={
+                    state.showAST === 'ts'
+                      ? astModel.storedTsAST
+                      : state.showAST === 'scope'
+                      ? astModel.storedScope
+                      : state.showAST === 'es'
+                      ? astModel.storedAST
+                      : undefined
+                  }
```
Comment (StyleShit) [inline packages/website/src/components/Playground.tsx:212]: maybe this is cleaner?
```suggestion
                  value={ {
                    ts: astModel.storedTsAST,
                    scope: astModel.storedScope,
                    es: astModel.storedAST,
                  }[state.showAST] }
                  
```

File: packages/website/src/components/Playground.module.css [hunk 3]
```diff
@@ -3,89 +3,95 @@
...
 
-.codeBlocks {
-  display: flex;
-  flex: 1 1 0;
-  flex-direction: row;
-  height: 100%;
-  width: calc(100vw - 20rem);
+.hidden {
+  display: none;
+  visibility: hidden;
```
Comment (StyleShit) [inline packages/website/src/components/Playground.module.css:19]: `display: none` doesn't render at all, why do you need `visibility: hidden`?

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: packages/website/src/components/Playground.module.css (hunk 3 above)
Comment (armano2) [reply packages/website/src/components/Playground.module.css:19]: we do not need it, i just moved this css from bellow

Comment (armano2) [reply packages/website/src/components/Playground.module.css:19]:
```suggestion
```

File: packages/website/src/components/Playground.tsx (hunk 2 above)
Comment (armano2) [reply packages/website/src/components/Playground.tsx:212]: agree, https://github.com/typescript-eslint/typescript-eslint/issues/7630

**Review by bradzacher — APPROVED** [review]
this is looking good to me from a design perspective and the react code sure looks like react code (it's been a while since I've properly written any react so :nodders:

my only point of feedback is this logic here -  chained, mixed logicals within a JSX expression are pretty gnarly to read, but we've already got #7630 for that so :nodders: again

**Comment by bradzacher** [comment]
something about this PR is breaking in CI. The initial install step is failing to finish (it was auto-cancelled after SIX hours)

**Review by JoshuaKGoldberg — APPROVED** [review]
Swell, thanks!

**Pull request merged**